File: flexpesa-client/scripts/test-auth.js
import axios from 'axios';

const API_BASE = 'http://localhost:8000/api/v1';

class FrontendAuthTester {
  constructor() {
    this.testUser = {
      email: 'frontend-test@example.com',
      password: 'testpassword123',
      first_name: 'Frontend',
      last_name: 'Test'
    };
    this.cookies = '';
  }

  log(message, type = 'info') {
    const colors = {
      info: '\x1b[36m',    // cyan
      success: '\x1b[32m', // green
      error: '\x1b[31m',   // red
      warning: '\x1b[33m'  // yellow
    };
    const reset = '\x1b[0m';
    console.log(`${colors[type]}[${type.toUpperCase()}]${reset} ${message}`);
  }

  async testBackendConnection() {
    this.log('🔗 Testing backend connection...', 'info');

    try {
      await axios.get(`${API_BASE}/../`);
      this.log('✅ Backend is running', 'success');
      return true;
    } catch (error) {
      this.log('❌ Backend not accessible. Start with: python run.py', 'error');
      return false;
    }
  }

  async testLogin() {
    this.log('🧪 Testing login...', 'info');

    try {
      const formData = new URLSearchParams();
      formData.append('username', 'admin@portfolio.com');
      formData.append('password', 'password');

      const response = await axios.post(`${API_BASE}/auth/cookie/login`, formData, {
        headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
        withCredentials: true
      });

      // Extract cookies
      const setCookieHeader = response.headers['set-cookie'];
      if (setCookieHeader) {
        this.cookies = setCookieHeader.join('; ');
      }

      this.log('✅ Login successful', 'success');
      return true;
    } catch (error) {
      this.log(`❌ Login failed: ${error.response?.data?.detail || error.message}`, 'error');
      return false;
    }
  }

  async testProtectedRoute() {
    this.log('🧪 Testing protected route...', 'info');

    try {
      const response = await axios.get(`${API_BASE}/portfolio/summary`, {
        headers: { 'Cookie': this.cookies },
        withCredentials: true
      });

      this.log('✅ Protected route accessible', 'success');
      this.log(`📊 Portfolio: ${response.data.accounts?.length || 0} accounts, $${response.data.total_value?.toFixed(2) || 0}`, 'info');
      return true;
    } catch (error) {
      this.log(`❌ Protected route failed: ${error.response?.status}`, 'error');
      return false;
    }
  }

  async runTests() {
    this.log('🚀 Starting Frontend Auth Tests', 'info');

    const tests = [
      () => this.testBackendConnection(),
      () => this.testLogin(),
      () => this.testProtectedRoute()
    ];

    let passed = 0;
    for (const test of tests) {
      if (await test()) passed++;
      await new Promise(resolve => setTimeout(resolve, 1000));
    }

    this.log(`🎯 Results: ${passed}/${tests.length} tests passed`, passed === tests.length ? 'success' : 'warning');
    return passed === tests.length;
  }
}

if (require.main === module) {
  const tester = new FrontendAuthTester();
  tester.runTests().then(success => process.exit(success ? 0 : 1));
}

File: flexpesa-ai/scripts/test_auth.py
#!/usr/bin/env python3
"""Test authentication system"""

import sys
import os

# Add project root to path
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from fastapi.testclient import TestClient
from app.main import app

def test_auth():
    """Test authentication endpoints"""
    client = TestClient(app)

    print("🧪 Testing authentication...")

    # Test registration
    print("1. Testing user registration...")
    response = client.post("/api/v1/auth/register", json={
        "email": "test@example.com",
        "password": "testpassword123",
        "first_name": "Test",
        "last_name": "User"
    })

    if response.status_code == 201:
        print("✅ Registration successful")
    else:
        print(f"⚠️ Registration: {response.status_code} - {response.text}")

    # Test login
    print("2. Testing login...")
    response = client.post("/api/v1/auth/cookie/login", data={
        "username": "admin@portfolio.com",
        "password": "password"
    })

    if response.status_code == 204:
        print("✅ Login successful")
        cookies = response.cookies
    else:
        print(f"❌ Login failed: {response.status_code} - {response.text}")
        return False

    # Test protected route
    print("3. Testing protected route...")
    response = client.get("/api/v1/portfolio/summary", cookies=cookies)

    if response.status_code == 200:
        print("✅ Protected route accessible")
        data = response.json()
        print(f"   Portfolio: {len(data.get('accounts', []))} accounts, ${data.get('total_value', 0):,.2f}")
    else:
        print(f"❌ Protected route failed: {response.status_code} - {response.text}")

    # Test unauthorized access
    print("4. Testing unauthorized access...")
    response = client.get("/api/v1/portfolio/summary")

    if response.status_code == 401:
        print("✅ Unauthorized access blocked")
    else:
        print(f"❌ Should be unauthorized: {response.status_code}")

    print("\n🎉 Authentication tests completed!")
    return True

if __name__ == "__main__":
    try:
        test_auth()
    except Exception as e:
        print(f"❌ Test failed: {e}")
        sys.exit(1)
        